### PR TITLE
[Bouffalolab] Update Dockerfile to setup compile environment for Bouffalo Lab SoC build

### PR DESCRIPTION
**Pull request by wy-hh**
Update Dockerfile to setup compile environment for Bouffalo Lab SoC build.

### CODE REVIEW

File: integrations/docker/images/chip-build-bouffalolab/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,11 @@
 ARG VERSION=latest
 FROM connectedhomeip/chip-build:${VERSION}
 
-RUN set -x \
-    && pip install bflb-iot-tool \
-    && apt-get update \
-    && apt-get install -fy \
-    vim \
-    && : # last line
+RUN apt update -y \
```
Comment (andy31415) [inline integrations/docker/images/chip-build-bouffalolab/Dockerfile:4]: since this changes dockerfile content (both bouffalolab and vscode) please update the version in chip-build/version with a new version and explanation on why the change was made.

File: integrations/docker/images/chip-build-bouffalolab/setup.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+set -x
+CURRENT_DIR=$(
+    cd "$(dirname "$0")"
+    pwd
+)
+SDK_ROOT_DIR=$CURRENT_DIR/../
```
Comment (andy31415) [inline integrations/docker/images/chip-build-bouffalolab/setup.sh:8]: you run this from `tmp` so SDK_ROOT_DIR will be `/`? Is this intended?

Should we maybe instead pass in the sdk as an argument to make the script easier to read? Then we could use /opt/sdk or similar as sdk paths.

File: integrations/docker/images/chip-build-vscode/Dockerfile [hunk 3]
```diff
@@ -50,6 +51,8 @@ COPY --from=ti /opt/ti/sysconfig_1.13.0 /opt/ti/sysconfig_1.13.0
 COPY --from=openiotsdk /opt/gcc-arm-none-eabi-10.3-2021.10/ /opt/gcc-arm-none-eabi-10.3-2021.10/
 COPY --from=openiotsdk /opt/FVP_Corstone_SSE-300/ /opt/FVP_Corstone_SSE-300/
 
+COPY --from=bouffalolab ~/.bouffalolab ~/.bouffalolab
```
Comment (andy31415) [inline integrations/docker/images/chip-build-vscode/Dockerfile:54]: why in $HOME? please use something like /opt like all the other toolchains.

File: integrations/docker/images/chip-build-bouffalolab/setup.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+TOOLCHAIN_SETUP_ROOT=~/.bouffalolab/toolchain
```
Comment (andy31415) [inline integrations/docker/images/chip-build-bouffalolab/setup.sh:10]: $HOME is not ideal. use something like `/opt/bouffallolab-sdk` or similar. The SDK is likely shared among all users of a system.

File: integrations/docker/images/chip-build-bouffalolab/setup.sh [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    path=${toolchains_url[i]}
+    url=${toolchains_url[i + 1]}
+    output=${toolchains_url[i + 2]}
+
+    wget -P "$TOOLCHAIN_SETUP_ROOT"/ "$url"
+    toolchain_zip=$(basename "$url")
+    if [ ! -f "$TOOLCHAIN_SETUP_ROOT/$toolchain_zip" ]; then
+        exit 1
+    fi
+    rm -rf "$TOOLCHAIN_SETUP_ROOT/$path" 2>&1 >/dev/null
```
Comment (andy31415) [inline integrations/docker/images/chip-build-bouffalolab/setup.sh:40]: why the hiding of all output here and below? It looks like we may want to see logs since this gets executed during docker image creation.

File: integrations/docker/images/chip-build-bouffalolab/Dockerfile (hunk 1 above)
Comment (wy-hh) [reply integrations/docker/images/chip-build-bouffalolab/Dockerfile:4]: Version and description updated

File: integrations/docker/images/chip-build-vscode/Dockerfile (hunk 3 above)
Comment (wy-hh) [reply integrations/docker/images/chip-build-vscode/Dockerfile]: Updated as your suggestion. Put it under /opt/bouffalolab_sdk.

File: integrations/docker/images/chip-build-bouffalolab/setup.sh (hunk 5 above)
Comment (wy-hh) [reply integrations/docker/images/chip-build-bouffalolab/setup.sh]: Updated.

File: integrations/docker/images/chip-build-bouffalolab/setup.sh (hunk 4 above)
Comment (wy-hh) [reply integrations/docker/images/chip-build-bouffalolab/setup.sh]: Updated.

File: integrations/docker/images/chip-build-bouffalolab/setup.sh (hunk 2 above)
Comment (wy-hh) [reply integrations/docker/images/chip-build-bouffalolab/setup.sh]: Script setup.sh is copied from the repo of Bouffalo Lab IOT SDK, and creates a symbolic link to installed toolchain under the repo for Bouffalo Lab examples building. SDK_ROOT_DIR is root folder of that repo. 

To be clear that TOOLCHAIN_SYMBOLIC_LINK_PATH is added for the path of symbolic link, and this symbolic link will not be created during Docker image building. Now, setup.sh will only download toolchain zip files and extract to /opt/bouffalolab_sdk.

**Review by andy31415 — APPROVED** [review]
